File: go-base-web/src/api/dict.ts
import request from '@/utils/request'
import type { ApiResponse } from '@/types'

// 字典类型
export interface DictType {
  id: number
  name: string
  type: string
  status: number
  remark: string
  created_at: string
  updated_at: string
}

// 字典数据
export interface DictData {
  id: number
  dict_type: string
  label: string
  value: string
  sort: number
  status: number
  tag_type: string
  is_default: number
  remark: string
  created_at: string
  updated_at: string
}

// 分页响应
export interface PageResponse<T> {
  list: T[]
  total: number
}

// ==================== 字典类型 ====================

// 获取字典类型列表（分页）
export function getDictTypeList(params?: { page?: number; page_size?: number; name?: string; type?: string; status?: number }) {
  return request.get<any, ApiResponse<PageResponse<DictType>>>('/dict/types', { params })
}

// 获取所有字典类型（不分页）
export function getAllDictTypes() {
  return request.get<any, ApiResponse<DictType[]>>('/dict/types/all')
}

// 获取字典类型详情
export function getDictType(id: number) {
  return request.get<any, ApiResponse<DictType>>(`/dict/types/${id}`)
}

// 创建字典类型
export function createDictType(data: { name: string; type: string; status?: number; remark?: string }) {
  return request.post<any, ApiResponse>('/dict/types', data)
}

// 更新字典类型
export function updateDictType(id: number, data: { name?: string; type?: string; status?: number; remark?: string }) {
  return request.put<any, ApiResponse>(`/dict/types/${id}`, data)
}

// 删除字典类型
export function deleteDictType(id: number) {
  return request.delete<any, ApiResponse>(`/dict/types/${id}`)
}

// ==================== 字典数据 ====================

// 获取字典数据列表（分页）
export function getDictDataList(params: { dict_type: string; page?: number; page_size?: number; label?: string; status?: number }) {
  return request.get<any, ApiResponse<PageResponse<DictData>>>('/dict/data', { params })
}

// 根据字典类型获取字典数据（不分页，用于下拉框）
export function getDictDataByType(type: string) {
  return request.get<any, ApiResponse<DictData[]>>(`/dict/type/${type}`)
}

// 获取字典数据详情
export function getDictData(id: number) {
  return request.get<any, ApiResponse<DictData>>(`/dict/data/${id}`)
}

// 创建字典数据
export function createDictData(data: {
  dict_type: string
  label: string
  value: string
  sort?: number
  status?: number
  tag_type?: string
  is_default?: number
  remark?: string
}) {
  return request.post<any, ApiResponse>('/dict/data', data)
}

// 更新字典数据
export function updateDictData(id: number, data: {
  label?: string
  value?: string
  sort?: number
  status?: number
  tag_type?: string
  is_default?: number
  remark?: string
}) {
  return request.put<any, ApiResponse>(`/dict/data/${id}`, data)
}

// 删除字典数据
export function deleteDictData(id: number) {
  return request.delete<any, ApiResponse>(`/dict/data/${id}`)
}

File: go-base-server/router/modules/dict.go
package modules

import (
	"github.com/gin-gonic/gin"

	v1 "go-base-server/api/v1"
	"go-base-server/model/request"
	"go-base-server/router/registry"
)

func init() {
	RegisterModule(&DictModule{})
}

type DictModule struct{}

func (m *DictModule) Name() string {
	return "字典管理"
}

func (m *DictModule) RegisterPublicRoutes(rg *gin.RouterGroup) {
	// 公开接口：根据类型获取字典数据（供下拉框等使用）
	R(rg, "GET", "/dict/type/:type", m.Name(), "获取字典数据", v1.Dict.GetDictDataByType)
}

func (m *DictModule) RegisterPrivateRoutes(rg *gin.RouterGroup) {
	// 字典类型
	R(rg, "GET", "/dict/types", m.Name(), "字典类型列表", v1.Dict.GetDictTypeList, registry.WithAuth())
	R(rg, "GET", "/dict/types/all", m.Name(), "所有字典类型", v1.Dict.GetAllDictTypes, registry.WithAuth())
	R(rg, "GET", "/dict/types/:id", m.Name(), "字典类型详情", v1.Dict.GetDictType, registry.WithAuth())
	R(rg, "POST", "/dict/types", m.Name(), "创建字典类型", v1.Dict.CreateDictType,
		registry.WithAuth(), registry.WithRequest(request.CreateDictTypeRequest{}))
	R(rg, "PUT", "/dict/types/:id", m.Name(), "更新字典类型", v1.Dict.UpdateDictType,
		registry.WithAuth(), registry.WithRequest(request.UpdateDictTypeRequest{}))
	R(rg, "DELETE", "/dict/types/:id", m.Name(), "删除字典类型", v1.Dict.DeleteDictType, registry.WithAuth())

	// 字典数据
	R(rg, "GET", "/dict/data", m.Name(), "字典数据列表", v1.Dict.GetDictDataList, registry.WithAuth())
	R(rg, "GET", "/dict/data/:id", m.Name(), "字典数据详情", v1.Dict.GetDictData, registry.WithAuth())
	R(rg, "POST", "/dict/data", m.Name(), "创建字典数据", v1.Dict.CreateDictData,
		registry.WithAuth(), registry.WithRequest(request.CreateDictDataRequest{}))
	R(rg, "PUT", "/dict/data/:id", m.Name(), "更新字典数据", v1.Dict.UpdateDictData,
		registry.WithAuth(), registry.WithRequest(request.UpdateDictDataRequest{}))
	R(rg, "DELETE", "/dict/data/:id", m.Name(), "删除字典数据", v1.Dict.DeleteDictData, registry.WithAuth())
}
